lib/cylc/job_file.py

```diff
@@ -206,6 +209,28 @@ def _write_environment_1(self, handle, job_conf):
             ' '.join(job_conf['namespace_hierarchy']))
         handle.write(
             '\n    export CYLC_TASK_TRY_NUMBER=%s' % job_conf['try_num'])
+        if job_conf['param_var']:
+                for var, val in job_conf['param_var'].items():
+                    value = str(val)  # (needed?)
+                    match = self.env_match1.match(value)
+                    if match:
+                        # ~foo/bar or ~/bar
+                        # write as ~foo/"bar" or ~/"bar"
+                        head, tail = match.groups()
+                        handle.write(
+                            '\n    export %s=%s"%s"' %
+                            (var, head, tail))
+                    elif self.env_match2.match(value):
+                        # plain ~foo or just ~
+                        # just leave unquoted as subsequent spaces don't
+                        # make sense in this case anyway
+                        handle.write('\n    export %s=%s' % (var, value))
+                    else:
+                        # Non tilde values - quote the lot.
+                        # This gets values like "~one ~two" too, but these
+                        # (in variable values) aren't expanded by the shell
+                        # anyway so it doesn't matter.
+                        handle.write('\n    export %s="%s"' % (var, value))
```

Comment:
@challurip - sorry, I didn't look closely enough first time through, but: (1) this is now the same code as in `_write_environment_1()` - where you cut-and-pasted it from - so it should be factored out into a function rather than repeated.  However, don't bother doing that because: (2) in fact we don't most of this code at all for parameter variables -  Unix file path `~` characters (as per the comments) can appear in general user-defined variables, but not in parameter values which are only simple strings or integers.  i.e. we only need the final simple case - the content under `else:`.